### ISSUE TITLE
empty metric_timestamp causes the iot topic rule to not be executed when using cloudwatch metrics as action

### ISSUE THREAD

**Issue by MattBlack85** [terraform-providers/terraform-provider-aws#6354]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
0.11.10, provider aws 1.29.0

### Affected Resource(s)

* aws_iot_topic_rule

### Terraform Configuration Files

```hcl
resource "aws_iot_topic_rule" "shadow_rejection_errors" {
 name = "shadow_rejected_error"
 description = "Add a metric to cloudwatch if any shadow fails update"
 enabled = true
 sql = "SELECT *, topic(3) as device_id FROM '$aws/things/+/shadow/update/rejected'"
 sql_version = "2016-03-23"

 cloudwatch_metric {
   metric_name = "Rejected"
   metric_namespace = "Shadow"
   metric_value = "1"
   metric_unit = "Count"
   role_arn = "${var.iot_metrics_to_cloudwatch_arn}"
 }
}
```

### Debug Output
https://gist.github.com/MattBlack85/d0f0a72811cb661a6acdce7f29717778

### Panic Output
None

### Expected Behavior
If a rule is created that will put a metric into cloudwatch and the optional parameter metric_timestamp is not provided, the rule should execute correctly with the default timestamp assigned by AWS

### Actual Behavior
When calling the AWS API an empty string is sent as `metric_timestamp`, this make AWS thinks that that empty string is the timestamp we want to use and the rule is not executed.
If the rule is created manually on the console it works and if we get the details from the AWS API we got:
```
{
    "ruleArn": "arn:aws:iot:eu-west-2:XXXXXXXX:rule/shadow_rejected_error",
    "rule": {
        "ruleName": "shadow_rejected_error",
        "sql": "SELECT *, topic(3) as device_id FROM '$aws/things/+/shadow/update/rejected'",
        "description": "Add a metric to cloudwatch if any shadow fails update",
        "actions": [
            {
                "cloudwatchMetric": {
                    "roleArn": "arn:aws:iam::XXXXXXX:role/iot_metrics_to_cloudwatch",
                    "metricNamespace": "Shadow",
                    "metricName": "Rejected",
                    "metricValue": "1",
                    "metricUnit": "Count"
                }
            }
        ],
        "ruleDisabled": false,
        "awsIotSqlVersion": "2016-03-23"
    }
}

```
If I create it using terraform the API returns:
```
{
    "ruleArn": "arn:aws:iot:eu-west-2:647260456021:rule/shadow_rejected_error",
    "rule": {
        "ruleName": "shadow_rejected_error",
        "sql": "SELECT *, topic(3) as device_id FROM '$aws/things/+/shadow/update/rejected'",
        "description": "Add a metric to cloudwatch if any shadow fails update",
        "actions": [
            {
                "cloudwatchMetric": {
                    "roleArn": "arn:aws:iam::647260456021:role/iot_metrics_to_cloudwatch",
                    "metricNamespace": "Shadow",
                    "metricName": "Rejected",
                    "metricValue": "1",
                    "metricUnit": "Count",
                    "metricTimestamp": ""
                }
            }
        ],
        "ruleDisabled": false,
        "awsIotSqlVersion": "2016-03-23"
    }
}
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
None

### References

* #0000

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.51.0 of the AWS provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.51.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1510-december-05-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
